coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -188,6 +203,17 @@ class QueryEngine(dataset: Dataset,
     }
   }
 
+  private def materializeMetadata(queryId: String,
+                                   submitTime: Long,
+                                   options: QueryOptions,
+                                   lp: Metadata): Seq[MetadataExecLeafPlan] = {
+    shardsFromFilters(lp.rawSeries.filters, options).map { shard =>
+      val dispatcher = dispatcherForShard(shard)
+      MetadataExecLeafPlan(queryId, submitTime, options.itemLimit, dispatcher, dataset.ref, shard,
+        lp.rawSeries.filters, lp.start, lp.end, lp.rawSeries.columns)
```

Comment:
Leaf plans should now take column IDs instead of strings, and validation of the column names to IDs should happen here in materialize.  Please see the latest code merged into develop for more details.